clash-lib/src/Clash/Driver/Manifest.hs

```diff
@@ -223,13 +239,7 @@ instance FromJSON Manifest where
         <*> (topComponent >>= (.: "ports_flat"))
         <*> v .: "components"
         <*> (topComponent >>= (.: "name"))
-        <*> do
-              files <- v .: "files"
-              forM files $ \obj -> do
-                fName <- obj .: "name"
-                sha256 <- obj .: "sha256"
-                -- See Note [Failed hex digest decodes]
-                pure (fName, unsafeFromHexDigest sha256)
+        <*> parseFiles v
```

Comment:
I think it might be better to change the field to a `FilesManifest` and then just use `parseJSON` here, that way if anything to do with how files + hashes are parsed you only need to change one place

Reply:
Makes sense